x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TimeseriesFieldTypeHelper.java

```diff
@@ -43,28 +40,21 @@ public boolean isTimeSeriesLabel(final String field, final Map<String, ?> unused
     public boolean isTimeSeriesMetric(final String unused, final Map<String, ?> fieldMapping) {
         final String metricType = (String) fieldMapping.get(TIME_SERIES_METRIC_PARAM);
         return metricType != null
-            && Arrays.asList(TimeSeriesParams.MetricType.values()).contains(TimeSeriesParams.MetricType.valueOf(metricType));
+            && List.of(TimeSeriesParams.MetricType.values()).contains(TimeSeriesParams.MetricType.valueOf(metricType));
     }
 
     public boolean isTimeSeriesDimension(final String unused, final Map<String, ?> fieldMapping) {
         return Boolean.TRUE.equals(fieldMapping.get(TIME_SERIES_DIMENSION_PARAM));
     }
 
     static class Builder {
-        private final IndicesService indicesService;
-        private final Map<String, Object> indexMapping;
-        private final IndexMetadata indexMetadata;
+        private final MapperService mapperService;
 
-        Builder(final IndicesService indicesService, final Map<String, Object> indexMapping, final IndexMetadata indexMetadata) {
-            this.indicesService = indicesService;
-            this.indexMapping = indexMapping;
-            this.indexMetadata = indexMetadata;
+        Builder(final MapperService mapperService) {
+            this.mapperService = mapperService;
         }
 
         public TimeseriesFieldTypeHelper build(final String timestampField) throws IOException {
-            final MapperService mapperService = indicesService.createIndexMapperServiceForValidation(indexMetadata);
-            final CompressedXContent sourceIndexCompressedXContent = new CompressedXContent(indexMapping);
-            mapperService.merge(MapperService.SINGLE_MAPPING_NAME, sourceIndexCompressedXContent, MapperService.MergeReason.INDEX_TEMPLATE);
```

Comment:
Was this important?

Reply:
Ah. It moved.

Reply:
Yes, I moved it one level up so that I can reuse the `MapperService` instance